sdk/python/feast/feature_store.py

```diff
@@ -1352,12 +1377,13 @@ def tqdm_builder(length):
                 tqdm_builder=tqdm_builder,
             )
 
-            self._registry.apply_materialization(
-                feature_view,
-                self.project,
-                start_date,
-                end_date,
-            )
+            if not isinstance(feature_view, OnDemandFeatureView):
```

Comment:
Since We are continuing the loop at line 1337&1338, we dont need this condition here.